libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -897,6 +897,38 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			m_context << success;
 			break;
 		}
+		case FunctionType::Kind::ABIEncode:
+		case FunctionType::Kind::ABIEncodePacked:
+		{
+			TypePointers argumentTypes;
+			for (auto const& arg: arguments)
+			{
+				arg->accept(*this);
+				argumentTypes.push_back(arg->annotation().type);
```

Comment:
I think you have to use something like `.type->mobileType()`.

Reply:
But perhaps that is done in `encodeToMemory`.

Reply:
This is a copy of the `SHA3` case above.